plugins/hls-explicit-imports-plugin/test/Main.hs

```diff
@@ -34,10 +34,43 @@ main = defaultTestRunner $ testGroup "import-actions"
     "Make imports explicit"
     [ codeActionAllGoldenTest "ExplicitUsualCase" 3 0
     , codeActionAllResolveGoldenTest "ExplicitUsualCase" 3 0
+    , inlayHintsTest "ExplicitUsualCase" 2 $ (@=?)
+        [InlayHint
+           { _position = Position {_line = 2, _character = 16}
+           , _label = InL "( a1 )"
+           , _kind = Just InlayHintKind_Type
+           , _textEdits = Just [TextEdit (Range (Position 2 0) (Position 2 16)) "import ExplicitA ( a1 )"]
+           , _tooltip = Nothing
+           , _paddingLeft = Just True
+           , _paddingRight = Nothing
+           , _data_ = Nothing
+           }]
     , codeActionOnlyGoldenTest "ExplicitOnlyThis" 3 0
     , codeActionOnlyResolveGoldenTest "ExplicitOnlyThis" 3 0
+    , inlayHintsTest "ExplicitOnlyThis" 3 $ (@=?)
+        [InlayHint
+           { _position = Position {_line = 3, _character = 16}
+           , _label = InL "( b1 )"
+           , _kind = Just InlayHintKind_Type
+           , _textEdits = Just [TextEdit (Range (Position 3 0) (Position 3 16)) "import ExplicitB ( b1 )"]
```

Comment:
These text edits are arguably overly aggressive. We _could_ just insert the export list where it needs to go, rather than replacing the entire import. That would have the advantage of preserving as much as possible of what the user wrote.

I'm not sure it makes a big difference, but we should think about it and write down why we chose this one. I think in general the expectation is that an inlay hint inserts just the text in the hint...

Reply:
Yes, append is what I did first, but I did run into some problems, like this code:
```haskell
import           Development.IDE                      hiding (pluginHandlers,
                                                       pluginRules)
```

If we just append it, it will obviously break the import statement.
So I chose the same behavior as the original code lens & code action, replacing the entire import statement, so that users only need to re-format it.

Reply:
Interesting! How does the inlay hint display in that case? I guess here the "perfect" thing to do would be to append the import list and also delete any "hiding" clauses. Complicated, though.

Reply:
Also, needs a comment, and maybe a test?

Reply:
> I guess here the "perfect" thing to do would be to append the import list and also delete any "hiding" clauses. Complicated, though.

The current approach is to _append inlay hints directly_ at the end.
I don't think it's possible to have inlay hints "cover" a part of your code.

> Also, needs a comment, and maybe a test?

Okay, I'll add both.

Reply:
>  I don't think it's possible to have inlay hints "cover" a part of your code.

Right, that's definitely true for the hint. I meant that the _text edits_ could only overwrite the hiding clause rather than the entire line.